highlighters/main/main-highlighter.zsh

```diff
@@ -664,12 +664,9 @@ _zsh_highlight_main_highlighter_highlight_list()
           next_word+=':start:'
           next_word+=':sudo_opt:'
         elif [[ $arg == '-'* ]]; then
-          # Unknown flag
+          # Unknown flag.  We don't know whether it takes an argument or not,
+          # so we don't modify $next_word.
           this_word=':sudo_opt:'
-          next_word+=':start:'
-          next_word+=':sudo_opt:'
-          _zsh_highlight_main_add_region_highlight $start_pos $end_pos unknown-token
-          continue
         else
```

Comment:
Discussed this on IRC with @phy1729. Our takeaway was to re-add the `next_word+=` settings.  That will fix #641.0 and #658's issue about unknown flags being red, and restore 0.6.0's behaviour of assuming unknown flags take no argument.  With that behaviour, flags that take no argument will highlight the inner command word correctly, and flags that take an argument will highlight the command word correctly if the argument is given in the same shell word as the flag (as in `-uphy1729` or `--user=phy1729` without spaces).  That's compared to the branch's current behaviour of never highlighting the inner command word at all.